### ISSUE TITLE
Rework user management UI to use new user creation/editing API

### ISSUE THREAD

**Issue by browndav** [medic/medic-webapp#2048]
Update the in-app user management UI to use the APIs built in #2046 and #2047.

Decisions from today's call: split security-sensitive fields from the `_users` database into its own modal dialog, including ordinary-user password change operations. We believe this will dramatically simplify document update conflict handling. It may also allow us to reconfirm a password/credentials for security-sensitive operations.

**Comment by browndav** [comment]
Blocked on #2046.

**Comment by mandric** [comment]
Minor security issue worth mentioning here because using the API provides us more leverage security-wise.

**Comment by nice-snek** [comment]
hi frand @browndav, please [triage](https://github.com/medic/medic-docs/blob/master/md/dev/workflow.md#triaging-old-issues) before the end of this sprint. :parrot_conga::parrot_conga::parrot_conga:

**Comment by browndav** [comment]
Closing and marking as ancient. Create a new issue if this or a similar feature/fix is needed.

**Comment by SCdF** [comment]
This was actually fixed as part of https://github.com/medic/medic-webapp/issues/3910
